app/models/location.rb

```diff
@@ -12,10 +25,16 @@ class Location < ActiveRecord::Base
   validates :organisation, presence: true
   validates :title, presence: true
   validates :address, presence: true
-  validates :phone, presence: true, if: ->(l) { l.booking_location.blank? }
-  validates :booking_location, presence: true, if: ->(l) { l.phone.nil? }
+  validates :booking_location, presence: { if: ->(l) { l.phone.nil? } }
   validates :version, presence: true
   validates :state, presence: true, inclusion: %w(old current)
+  validates :phone,
+            presence: { if: ->(l) { l.booking_location.blank? } },
+            '::UkPhoneNumber' => { if: ->(l) { l.phone.present? && l.current? } }
```

Comment:
It might make sense to put those two conditionals behind a descriptively named method, so it calls out that behaviour. Here and elsewhere below. You could also then just pass the method as `if: :your_predicate?` without the lambda and args.